### PR TITLE
add support for google_storage_bucket_iam_binding

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #1065
| ❓ Documentation  | https://github.com/cloudskiff/driftctl-docs/pull/131

## Description

add support for google_storage_bucket_iam_binding

### CODE REVIEW

**Review by sundowndev — CHANGES_REQUESTED** [review]

File: pkg/remote/google_storage_scanner_test.go [hunk 1]
```diff
@@ -156,3 +157,140 @@ func TestGoogleStorageBucket(t *testing.T) {
...
+	providerVersion := "3.78.0"
+	resType := resource.ResourceType(googleresource.GoogleStorageBucketIamBindingResourceType)
+	schemaRepository := testresource.InitFakeSchemaRepository("google", providerVersion)
+	googleresource.InitResourcesMetadata(schemaRepository)
+	factory := terraform.NewTerraformResourceFactory(schemaRepository)
+	deserializer := resource.NewDeserializer(factory)
+
+	for _, c := range cases {
+		t.Run(c.test, func(tt *testing.T) {
+			repositoryCache := cache.New(100)
```
[the remark this answers is not in the document]
Comment (moadibfr) [reply pkg/remote/google_storage_scanner_test.go:273]: we are using mock for repository unless there is an update needed (this is why there is a cache, and we don't really need to use a mock for this usage)

[the remark this answers is not in the document]
Comment (sundowndev) [reply pkg/remote/google_storage_scanner_test.go:273]: you're right, my bad

File: pkg/remote/google_storage_scanner_test.go [hunk 2]
```diff
@@ -156,3 +158,172 @@ func TestGoogleStorageBucket(t *testing.T) {
...
+				c.storageRepositoryMock(storageRepo)
+			}
+			var storageRepository repository.StorageRepository = storageRepo
+			storageClient, err := storage.NewClient(context.Background())
+			if shouldUpdate {
+				if err != nil {
+					panic(err)
+				}
+				storageRepository = repository.NewStorageRepository(storageClient, repositoryCache)
+			}
```
Comment (sundowndev) [inline pkg/remote/google_storage_scanner_test.go:298]: why isn't storageClient initialized in the `shouldUpdate` condition ?

File: pkg/middlewares/google_legacy_bucket_iam_bindings_test.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		expected           []*resource.Resource
+	}{
+		{
+			"test that non legacy bindings are not ignored when managed by IaC",
+			[]*resource.Resource{
+				{
+					Id: "fake",
+				},
+				{
+					Id:   "admin bucket",
```
[the remark this answers is not in the document]
Comment (sundowndev) [reply pkg/middlewares/google_legacy_bucket_iam_bindings_test.go]: field names are still missing

**Review by sundowndev — CHANGES_REQUESTED** [review]
conflicts to resolve

File: pkg/resource/google/google_storage_bucket_iam_binding.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+	resourceSchemaRepository.SetNormalizeFunc(GoogleStorageBucketIamBindingResourceType, func(res *resource.Resource) {
+		res.Attributes().SafeDelete([]string{"force_destroy"})
+		res.Attributes().SafeDelete([]string{"etag"})
+	})
+	resourceSchemaRepository.SetResolveReadAttributesFunc(GoogleStorageBucketIamBindingResourceType, func(res *resource.Resource) map[string]string {
+		return map[string]string{
+			"bucket": *res.Attrs.GetString("bucket"),
+		}
+	})
+	resourceSchemaRepository.SetFlags(GoogleStorageBucketIamBindingResourceType, resource.FlagDeepMode)
```
Comment (sundowndev) [inline pkg/resource/google/google_storage_bucket_iam_binding.go:17]: either this resource shouldn't have the deep mode, or you forgot to add a details fetcher for it

**Review by sundowndev — CHANGES_REQUESTED** [review]

File: pkg/resource/google/google_storage_bucket_iam_binding.go (hunk 4 above)
Comment (moadibfr) [reply pkg/resource/google/google_storage_bucket_iam_binding.go:17]: no it doesn't need the details fetcher the details are all already fetched on enumeration. And calling the readresource remove the member list for a reason I couldn't find

Comment (sundowndev) [reply pkg/resource/google/google_storage_bucket_iam_binding.go:17]: I can't find any docs about that resource, is that related to a particular version of the provider ?

Comment (sundowndev) [reply pkg/resource/google/google_storage_bucket_iam_binding.go:17]: Ok figured out it's named resource_iam_binding in the provider's code https://github.com/hashicorp/terraform-provider-google/blob/d696c5ed2fb052896b29a6502f1f7b3b2d1c650d/google/resource_iam_binding.go#L65

Comment (moadibfr) [reply pkg/resource/google/google_storage_bucket_iam_binding.go:17]: yeah there is se same 3 kind of iam resources for plenty of service so they made a generic way to handle them.
But still I made the correct details fetcher and it didn't give the member list. And as we already have everything I removed it...

Comment (sundowndev) [reply pkg/resource/google/google_storage_bucket_iam_binding.go:17]: You forgot to add the `role` attribute to read attributes (see https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_iam_binding.go#L281-L291), I tested on your branch and it no longer drifts with the common details fetcher

**Review by sundowndev — APPROVED** [review]

**Pull request merged**